java/src/jmri/jmrix/openlcb/swing/stleditor/StlEditorPane.java

```diff
@@ -687,85 +723,250 @@ private boolean isTimerVar(String name) {
+            _tokenMap.put(matchComment1.start(), new Token("Comment", comment, matchComment1.start(), matchComment1.end()));
+        }
+
+        var matchComment2 = PARSE_COMMENT2.matcher(line);
+        while (matchComment2.find()) {
+            var comment = matchComment2.group(1).trim();
+            _tokenMap.put(matchComment2.start(), new Token("Comment", comment, matchComment2.start(), matchComment2.end()));
+        }
+
+        // Check for overlapping jump destinations and following labels
+        for (Token token : _tokenMap.values()) {
+            if (token.getType().equals("Dest")) {
+                var nextKey = _tokenMap.higherKey(token.getStart());
+                if (nextKey != null) {
+                    var nextToken = _tokenMap.get(nextKey);
+                    if (nextToken.getType().equals("Label")) {
+                        if (token.getEnd() > nextToken.getStart()) {
+                            _messages.add(Bundle.getMessage("ErrDestLabel", token.getName(), nextToken.getName()));
                         }
                     }
                 }
             }
+        }
+
+        if (_messages.size() > 0) {
+            // Display messages
+            String msgs = _messages.stream().collect(java.util.stream.Collectors.joining("\n"));
+            JmriJOptionPane.showMessageDialog(null,
+                    Bundle.getMessage("MsgParseErr", group.getName(), msgs),
+                    Bundle.getMessage("TitleParseErr"),
+                    JmriJOptionPane.ERROR_MESSAGE);
+            _messages.forEach((msg) -> {
+                log.error(msg);
+            });
+        }
+
+        // Create token debugging output
+        if (log.isDebugEnabled()) {
+            log.info("Line = {}", line);
+            for (Token token : _tokenMap.values()) {
+                log.info("Token = {}", token);
+            }
+        }
+    }
 
-            var logic = new LogicRow(label, oper, name, comment);
-            groupRow.getLogicList().add(logic);
+    /**
+     * Starting as the operator location minus one, work backwards to find a valid operator. When
+     * one is found, create and return the token object.
+     * @param index The current location in the line.
+     * @param line The line for the current group.
+     * @return a token or null.
+     */
+    private Token findOperator(int index, String line) {
+        var sb = new StringBuilder();
+        int limit = 10;
+
+        while (limit > 0) {
+            var ch = line.charAt(index);
+            if (ch != ' ') {
+                sb.insert(0, ch);
+                if (getEnum(sb.toString()) != null) {
+                    String oper = sb.toString();
+                    return new Token("Oper", oper, index, index + oper.length());
+                }
+            }
+            limit--;
+            index--;
         }
+        _messages.add(Bundle.getMessage("ErrNoOper", index, line));
+        return null;
     }
```

Comment:
Just checking. I don't fully understands this code, what it's supposed to do. But I have a question. If `line` doesn't have a valid operator (maybe due to bad user input) and `index` is <= 8, then it seems to me that line 946 will eventually throw an exception due to `line.charAt(-1)`.

Maybe the while loop a line 945 should also check that index >= 0.

Reply:
A typical STL line:  "AN I1.0".  The I1.0 variable is a consistent format that can be found using regex.  The logic goes backwards to find the operator, in this case AN which is a boolean AND NOT.  

Thanks for catching that.